src/lib/formats/tzx_cas.cpp

```diff
@@ -791,11 +900,20 @@ static int cdt_cas_fill_wave( int16_t *buffer, int length, uint8_t *bytes )
 {
 	int16_t *p = buffer;
 	int size = 0;
-	t_scale = (40 / 35);  /* scale to 4MHz */
```

Comment:
Where are you setting the scale for CDT format after your change?

Reply:
I have quite a few doubts about this part of the code, since the creation of the pulses is done before that part so I don't know if it is actually used. I give value 1 when it is going to create the spectrum blocks and value 1.44 when it is going to create a block 4b of msx. The Amstrad CDTs use spectrum blocks so they use the value 1, I have verified that both the TZX in spectrum and the CDT in Amstrad work.

At the beginning of the original code it is defined with the value 1 supposedly by the 4Mhz of Amstrad
(58 static float t_scale = 1; / * for scaling T-states to the 4MHz CPC * /)
but later in the line that you comment it is given value (40/35), that's why I say that I don't understand it.

Reply:
If the tape waveforms are created by software, wouldn’t the frequencies depend on the CPU speeds?

Reply:
I just read the Spectrum and Amstrad specs and they both have a Z80 at 4 Mzh, while the MSX goes at 3.58mhz, so it seems to be correct that Amstrad and Spectrum use the same t_scale value. I calculated the MSX t_scale value with the size of the pulses that come in the TSX and with the original wav values, which is correct because I have checked it with all types of load.

Reply:
I’m still not comfortable with this behaviour changing with addition of a completely unrelated feature.  There must be some rationale for it being here.